crates/turborepo-lib/src/run/mod.rs

```diff
@@ -131,6 +138,31 @@ impl Run {
         )?;
 
         info!("created cache");
+        let engine = EngineBuilder::new(
+            &self.base.repo_root,
+            &pkg_dep_graph,
+            opts.run_opts.single_package,
+        )
+        .with_root_tasks(root_turbo_json.pipeline.keys().cloned())
+        .with_turbo_jsons(Some(
+            Some((WorkspaceName::Root, root_turbo_json.clone()))
+                .into_iter()
+                .collect(),
+        ))
+        .with_tasks_only(opts.run_opts.only)
+        .with_workspaces(
+            filtered_pkgs
```

Comment:
Eventually, I'd like for `filtered_pkgs` to be a set of `WorkspaceName` instead of `Strings`.